packages/chevrotain/src/scan/lexer.ts

```diff
@@ -881,6 +881,21 @@ export function performRuntimeChecks(
               `<${currModeName}> at index: <${currIdx}>\n`,
             type: LexerDefinitionErrorType.LEXER_DEFINITION_CANNOT_CONTAIN_UNDEFINED
           })
+        } else if (has(currTokType, "LONGER_ALT")) {
+          const longerAlt = isArray(currTokType.LONGER_ALT)
+            ? currTokType.LONGER_ALT
+            : [currTokType.LONGER_ALT]
+          forEach(longerAlt, (currLongerAlt) => {
+            if (
+              !isUndefined(currLongerAlt) &&
+              !includes(currModeValue, currLongerAlt)
```

Comment:
can LONGER_ALT be combined with token categories? 
if so this `includes` check may not suffice

Reply:
It can be combined with token categories, though I'm not sure what we should actually validate there. If for example `modeA` contains category `Keyword` and another token which uses `Keyword`, but all tokens belonging to the `Keyword` category are in another mode `modeB`, this will simply lead to a managed lexer error. I believe this is expected behavior.

This PR is trying to explicitly prevent the uncaught error that appears in the scenario outlined in the issue. Though I'll add a test case for the scenario which I've just outlined above to confirm that it does what I think it does.

Reply:
From what I see a LONGER_ALT cannot be a token category
- https://github.com/Chevrotain/chevrotain/blob/master/packages/chevrotain/src/scan/lexer_public.ts#L570-L598

In which case this check should be fine.

Reply:
Alright, makes sense.